libraries/AP_Math/location.cpp

```diff
@@ -259,6 +259,21 @@ bool location_sanitize(const struct Location &defaultLoc, struct Location &loc)
         loc.alt = defaultLoc.alt;
         has_changed = true;
     }
+
+    // limit lat/lng to appropriate ranges
+    if (abs(loc.lat) > 90 * 1e7 || abs(loc.lng) > 180 * 1e7) {
+        loc.lat = defaultLoc.lat;
+        loc.lng = defaultLoc.lng;
+        has_changed = true;
+    }
+
+    if (abs(loc.alt) >= 0x7fffff) {
```

Comment:
this won't do anything as loc.alt is 24 bit, so this if() is never true.
in our discussion on checking this I presumed the check would happen on the value in the command before it was in the Location structure. Once it is in the Location structure its too late to tell if its out of range